pylint/checkers/stdlib.py

```diff
@@ -466,27 +466,36 @@ class StdlibChecker(DeprecatedMixin, BaseChecker):
 
     def __init__(self, linter: Optional["PyLinter"] = None) -> None:
         BaseChecker.__init__(self, linter)
+        self._deprecated_names_populated = False
         self._deprecated_methods: Set[Any] = set()
+        self._deprecated_attributes: Dict = {}
+        self._deprecated_classes: Dict = {}
+        self._deprecated_modules: Set[Any] = set()
+        self._deprecated_decorators: Set[Any] = set()
```

Comment:
```suggestion
        self._deprecated_decorators: Set[Any] = set()
        self._ensure_deprecated_names_populated()
```
Would that work instead of calling it in each class function?

Reply:
Unfortunately no, since`__init__()` runs before all option providers are parsed.

**Suggestion**
```
% python3.9 -m pylint tests/functional/d/deprecated/deprecated_methods_py39.py --py-version=3.8
************* Module deprecated_methods_py39
tests/functional/d/deprecated/deprecated_methods_py39.py:4:0: W1505: Using deprecated method b2a_hqx() (deprecated-method)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```
**PR**
```
% python3.9 -m pylint tests/functional/d/deprecated/deprecated_methods_py39.py --py-version=3.8

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 5.00/10, +5.00)
```

Reply:
Added a test case 9822f2f2eb66a9ebe596c8a5124d1d33e07bfa73

Reply:
You can use the `open` method for it.
https://github.com/PyCQA/pylint/blob/6c0ba130730e82fc21063f138ffd05347bea0b9e/pylint/extensions/typing.py#L143-L144